### PR TITLE
Query commit list using branch filter for benchmark history

### PR DESCRIPTION
**Pull request by keyonghan**
This PR fixes: https://github.com/flutter/flutter/issues/52694

### CODE REVIEW

File: app_dart/lib/src/service/datastore.dart [hunk 1]
```diff
@@ -89,8 +89,6 @@ class DatastoreService {
   }
 
   // Queries for recent commits without considering branches.
-  // TODO(keyonghan): combine this function with the above `queryRecentCommits`,
-  // this needs to fix https://github.com/flutter/flutter/issues/52694.
   Stream<Commit> queryRecentCommitsNoBranch({int limit = 100, int timestamp}) {
```
Comment (digiter) [inline app_dart/lib/src/service/datastore.dart:92]: Is it ready to remove this function everywhere?

Comment (keyonghan) [reply app_dart/lib/src/service/datastore.dart:92]: Not yet, task reservation is still using it: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/service/task_provider.dart#L30

Comment (digiter) [reply app_dart/lib/src/service/datastore.dart:92]: Can we remove that one altogether?

Comment (keyonghan) [reply app_dart/lib/src/service/datastore.dart:92]: Actually it is not a simple removal thing. As for task reservation, we need to treat all tasks equally no matter which branch it is, and reserve based on only time. So, we need to keep the NoBranch function at this moment.

**Pull request merged**